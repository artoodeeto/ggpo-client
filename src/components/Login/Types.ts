export enum LoginTypes {
  LOGIN = 'LOGIN',
  LOGOUT = 'LOGOUT'
}
