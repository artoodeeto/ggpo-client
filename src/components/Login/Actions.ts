import { LoginTypes } from './Types';

export const logMein = (id: number) => ({
  type: LoginTypes.LOGIN,
  payload: { id }
});
