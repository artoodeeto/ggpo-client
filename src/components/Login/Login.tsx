import React from 'react';
// import LoginStyles from './App.module.scss';

function testSubmit(event: any) {
  event.preventDefault();
  console.log(event);
}

class Login extends React.Component {
  render() {
    return (
      <div>
        <div>
          <form onSubmit={testSubmit}>
            <label htmlFor="email">
              Username:
              <input placeholder="youremail@address.com" type="text" name="email" id="" required />
            </label>

            <label htmlFor="password">
              Password:
              <input placeholder="password" type="text" name="email" id="" required />
            </label>

            <input type="submit" value="Login" />
          </form>
        </div>
      </div>
    );
  }
}

export default Login;
