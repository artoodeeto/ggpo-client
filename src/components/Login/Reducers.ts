import { LoginTypes } from './Types';

const initialState = {
  isAuthenticated: false
};

const loginUser = (state = initialState, action) => {
  switch (action.type) {
    case LoginTypes.LOGIN: {
      return {
        ...state,
        isAuthenticated: true
      };
    }
    default: {
      return state;
    }
  }
};

export default loginUser;
