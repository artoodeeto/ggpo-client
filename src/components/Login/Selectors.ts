export const isUserAuthorized = (store: any) => store.user.isAuthenticated;
