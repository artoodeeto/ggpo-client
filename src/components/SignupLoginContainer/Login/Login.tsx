import React, { FormEvent, useState, useEffect } from 'react';
import { LoginSignUp } from 'interfaces/signupLogin';
import { connect } from 'react-redux';
import * as userActions from '../Store/Actions';
import * as userSelectors from '../../../models/User/selectors';
// import LoginStyles from './App.module.scss';
import { State } from '../../../interfaces/stateInterface';
import { useHistory } from 'react-router-dom';

const onSubmitLogin = (event: FormEvent, email: string, password: string, onLogin: Function): void => {
  event.preventDefault();

  const formSignup: LoginSignUp = {
    email,
    password
  };

  onLogin(formSignup);
};

const Login = (props: any) => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const history = useHistory();

  useEffect(() => {
    if (props.isAuthenticated) history.push('/feed');
  });

  return (
    <div>
      <div>
        <h1>{props.isAuthenticated ? props.foo : 'nope'}</h1>
        <button onClick={props.onLogout}>LOGOUT</button>
        <form onSubmit={(e) => onSubmitLogin(e, email, password, props.onLogin)}>
          <label htmlFor="email">
            Email:
            <input
              onChange={(e) => setEmail(e.target.value)}
              value={email}
              placeholder="youremail@address.com"
              type="text"
              name="email"
              id=""
              required
            />
          </label>

          <label htmlFor="password">
            Password:
            <input
              onChange={(e) => setPassword(e.target.value)}
              value={password}
              placeholder="password"
              type="text"
              name="password"
              id=""
              required
            />
          </label>

          <input type="submit" value="Login" />
        </form>
      </div>
    </div>
  );
};

const mapStateToProps = (state: State) => {
  return {
    foo: 'yep',
    isAuthenticated: userSelectors.isUserAuthorized(state),
    userInfo: userSelectors.userInfo(state)
  };
};

const mapDispatchToProps = (dispatch: Function) => {
  return {
    onLogin: (userLogin: LoginSignUp) => dispatch(userActions.logMeIn(userLogin)),
    onLogout: () => dispatch(userActions.logMeOut())
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Login);
