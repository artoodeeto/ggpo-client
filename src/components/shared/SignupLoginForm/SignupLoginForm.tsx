import React, { FC, FormEvent, useState } from 'react';
import { ILoginSignUpFormParams } from 'interfaces/session';
import { connect } from 'react-redux';
import { AnyAction } from 'redux';
import { ThunkDispatch } from 'redux-thunk';
import { RootState } from 'store/root/root_reducer';
import { logMeIn, signMeUp } from 'store/session/Actions';
import styles from './SignupLoginForm.module.scss';

const handleSubmit = (
  event: FormEvent,
  params: ILoginSignUpFormParams,
  signupOrLoginAPI: (userSignup: ILoginSignUpFormParams) => void
): void => {
  event.preventDefault();

  signupOrLoginAPI(params);
};

type SignupLoginFormProps = {
  toLoginOrSignup: boolean;
  onLogin: (userLogin: ILoginSignUpFormParams) => void;
  onSignup: (userSignup: ILoginSignUpFormParams) => void;
};

// TODO: change this as array of objects so we wont have render each input. we'll replace this with filter.
/**
 * TODO: sample object
 * [{
 *  onchange: fn,
 *  value,
 *  placeHolder,
 * type
 * }]
 */
export const SignupLoginForm: FC<SignupLoginFormProps> = ({ toLoginOrSignup, onLogin, onSignup }) => {
  const [email, setEmail] = useState('');
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');
  const [confirmPassword, setConfirmPassword] = useState('');
  const params = {
    email,
    username,
    password
  };

  return (
    <form
      className={styles.SignUpLoginForm}
      onSubmit={(e) => handleSubmit(e, params, toLoginOrSignup ? onLogin : onSignup)}
    >
      <input
        onChange={(e) => setEmail(e.target.value)}
        value={email}
        placeholder="email_address@gmail.com"
        type="text"
        name="email"
        id=""
        required
      />

      {toLoginOrSignup ? null : (
        <input
          onChange={(e) => setUsername(e.target.value)}
          value={username}
          placeholder="user name"
          type="text"
          name="username"
          id=""
          required
        />
      )}

      <input
        onChange={(e) => setPassword(e.target.value)}
        value={password}
        placeholder="password"
        type="text"
        name="password"
        id=""
        required
      />

      {toLoginOrSignup ? null : (
        <input
          onChange={(e) => setConfirmPassword(e.target.value)}
          value={confirmPassword}
          placeholder="confirm password"
          type="text"
          name="conf_password"
          id=""
          required
        />
      )}

      <input type="submit" value={toLoginOrSignup ? 'Login' : 'Signup'} />
    </form>
  );
};

const mapStateToProps = (state: RootState) => {
  return {};
};

const mapDispatchToProps = (dispatch: ThunkDispatch<RootState, {}, AnyAction>) => {
  return {
    onSignup: (userSignup: ILoginSignUpFormParams) => dispatch(signMeUp(userSignup)),
    onLogin: (userLogin: ILoginSignUpFormParams) => dispatch(logMeIn(userLogin))
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(SignupLoginForm);
