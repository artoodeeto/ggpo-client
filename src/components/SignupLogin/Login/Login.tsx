import React, { FormEvent, useState, FC } from 'react';
import { connect } from 'react-redux';
// import LoginStyles from './App.module.scss';
import { ILoginSignUpFormParams } from 'interfaces/session';
import { showLoading } from 'store/session/Selectors';
import { logMeIn } from 'store/session/Actions';
import { RootState } from 'store/root/root_reducer';
import { AnyAction } from 'redux';
import { ThunkDispatch } from 'redux-thunk';

type LoginProps = {
  toShowLoading: boolean;
  onLogin: (userLogin: ILoginSignUpFormParams) => void;
};

const onSubmitLogin = (
  event: FormEvent,
  email: string,
  password: string,
  onLogin: (userLogin: ILoginSignUpFormParams) => void
): void => {
  event.preventDefault();

  const formSignup: ILoginSignUpFormParams = {
    email,
    password
  };
  onLogin(formSignup);
};

const Login: FC<LoginProps> = ({ toShowLoading, onLogin }) => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  return (
    <div>
      <div>
        <div>{toShowLoading ? 'LOGGING EN' : ''}</div>
        <form onSubmit={(e) => onSubmitLogin(e, email, password, onLogin)}>
          <label htmlFor="email">
            Email:
            <input
              onChange={(e) => setEmail(e.target.value)}
              value={email}
              placeholder="youremail@address.com"
              type="text"
              name="email"
              id=""
              required
            />
          </label>

          <label htmlFor="password">
            Password:
            <input
              onChange={(e) => setPassword(e.target.value)}
              value={password}
              placeholder="password"
              type="text"
              name="password"
              id=""
              required
            />
          </label>

          <input type="submit" value="Login" />
        </form>
      </div>
    </div>
  );
};

const mapStateToProps = (state: RootState) => {
  return {
    toShowLoading: showLoading(state)
  };
};

const mapDispatchToProps = (dispatch: ThunkDispatch<RootState, {}, AnyAction>) => {
  return {
    onLogin: (userLogin: ILoginSignUpFormParams) => dispatch(logMeIn(userLogin))
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Login);
