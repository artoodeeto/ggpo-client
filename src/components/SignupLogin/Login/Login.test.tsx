import React from 'react';
import SignupLoginForm from 'components/shared/SignupLoginForm/SignupLoginForm';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Login } from './Login';

describe('<Login/>', () => {
  describe('Login', () => {
    it('should show SignupLoginForm with props', () => {
      const wrapper = shallow(<Login isLoggingInOrSigningUp={false} />);
      expect(shallowToJson(wrapper)).toMatchSnapshot();

      expect(wrapper.find(SignupLoginForm).exists()).toBeTrue();
      expect(wrapper.props()).toContainKeys(['toLoginOrSignup']);
    });
  });
});
