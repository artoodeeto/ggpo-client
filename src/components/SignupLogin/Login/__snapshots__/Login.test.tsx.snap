// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login/> Login should show SignupLoginForm with props 1`] = `
<Connect(SignupLoginForm)
  toLoginOrSignup={false}
/>
`;
